staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package bootstrap provides a token authenticator for TLS bootstrap secrets.
+*/
+package bootstrap
+
+import (
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+const (
+	SecretType = "bootstrap.kubernetes.io/token"
+
+	TokenID     = "token-id"
+	TokenSecret = "token-secret"
+
+	BootstrapUserPrefix = "system:bootstrap:"
+	BootstrapGroup      = "system:bootstrappers"
+)
+
+// SecretLister is a Kubernetes clients which can list secrets in a namespace.
+type SecretLister interface {
```

Comment:
Could we instead do something like this so it's possible to pass a client-go ClientSet to it for example?
```go
// SecretsGetter has a method to return a SecretInterface.
// A group's client should implement this interface.
type SecretsGetter interface {
	Secrets(namespace string) SecretInterface
}

// SecretInterface has methods to work with Secret resources.
type SecretInterface interface {
	List(opts meta_v1.ListOptions) (*v1.SecretList, error)
}
```
(taken from https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/secret.go)
Maybe the client-go clientset would do the caching for us then?